graphene-reader/src/main/java/net/iponweb/disthene/reader/format/ResponseFormatter.java

```diff
@@ -195,4 +179,46 @@ private static void consolidate(List<TimeSeries> timeSeriesList, int maxDataPoin
 
         return result;
     }
+
+    private static int computeStringBuilderCapacity(List<TimeSeries> timeSeriesList) {
+        // For each TimeSeries, 28 = {"target":,"datapoints":[]},
+        // Subtract 1 because the last TimeSeries does not have comma.
+        // Plus 2 = [] --> 28 * timeSeriesList.size() - 1 + 2 = 28 * timeSeriesList.size() + 1
+        int stringBuilderCapacity = 28 * timeSeriesList.size() + 1;
+        for (TimeSeries ts : timeSeriesList) {
+            ts.setName(GSON.toJson(ts.getName()));
+            // For each timeSeries, there are
+            // 1. TimeSeries name length
+            // 2. Length of timestamp = 10
+            // 3. Necessary symbols = [,], = 4
+            // So, 1 + (2 + 3) * (Count of values), but subtract 1 because the last datapoint does not have comma.
+            stringBuilderCapacity += ts.getName().length() + (14 * ts.getValues().length) - 1;
+            // Count for each datapoint's length.
+            for (Double val : ts.getValues()) {
+                stringBuilderCapacity += String.valueOf(val).length();
+            }
```

Comment:
I think, this logic possibly changes to calculate about max length like below:
```capacity += (Double max byte size) * ts.getValues().size()```

Reply:
I don't know the actual upper bound but this is the good point.
By following the article(https://stackoverflow.com/questions/21146544/what-is-the-maximum-length-of-double-tostringd), we can use 26 instead.
The only thing that I'm concerning is when system encounters edge case.
For example, let's assume the actual length of each datapoint is 1. If we use upper bound then the system will calculate 25 * ts.getValues().size() more than the actual and this may lead the system to calculate capacity exceeds Integer MAX as the number of datapoint is getting larger. In this case the system cannot make response even if it can because the limit of StringBuilder's capacity is Integer MAX.